common-custom-user-data-gradle-plugin/src/main/java/com/gradle/SystemPropertyOverrides.java

```diff
@@ -53,7 +53,8 @@ void configureBuildCache(BuildCacheConfiguration buildCache) {
         });
 
         // null check required to avoid creating a remote build cache instance when none was already present in the build
-        if (buildCache.getRemote() != null) {
+        // We need to ensure that this is the HttpBuildCache and not another build cache type like AWS S3.
+        if (buildCache.getRemote() != null && buildCache.getRemote() instanceof HttpBuildCache) {
```

Comment:
Can remove null check which will be handled by instanceof

Reply:
done, thanks.